frontend/online-shop/src/App.tsx

```diff
@@ -0,0 +1,26 @@
+import "./App.css";
+import { TableProducts } from "./components/Products";
+import { Switch, Route, Redirect } from "react-router-dom";
+import ProductDetails from "./components/ProductDetails";
+import { ThemeProvider } from "@material-ui/core";
+import theme from "./styles/tableTheme";
+import Header from "./Header";
+
+export const App = () => {
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <Header />
+        <main>
+          <Switch>
+            <Route exact path="/">
+              <Redirect to="/products" />
+            </Route>
+            <Route path="/product/:number" component={ProductDetails} />
```

Comment:
You should be more specific with the path parameters. Either /product/:id or /product/:productId.

Reply:
Now its more specific